Source/Core/Common/Debug/OSThread.cpp

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Common/Debug/OSThread.h"
+
+#include <algorithm>
+#include "fmt/format.h"
+
+#include "Core/PowerPC/MMU.h"
+
+// Context offsets based on the following functions:
+//  - OSSaveContext
+//  - OSSaveFPUContext
+//  - OSDumpContext
+//  - OSClearContext
+//  - OSExceptionVector
+void Common::Debug::OSContext::Read(u32 addr, OSContext* ctx)
+{
+  for (int i = 0; i < 32; i++)
```

Comment:
This loop and the ones below it can just query the array's size instead of hardcoding the value. `int` shouldn't be used here either since the functions expect an address, so this will cause sign mismatch warnings.